src/Storages/StorageMergeTree.cpp

```diff
@@ -1283,7 +1296,8 @@ MergeMutateSelectedEntryPtr StorageMergeTree::selectPartsToMutate(
             return std::make_shared<MergeMutateSelectedEntry>(future_part, std::move(tagger), commands, txn);
         }
     }
-
+    if (exist_posponed_failed_part)
+        mutation_wait_event.notify_all();
```

Comment:
All other usages of `mutation_wait_event` notify it under
`std::lock_guard lock(mutation_wait_mutex)`. Generally you should not notify a condition variable while holding a mutex (as waiting threads will be notified just to sleep again) but maybe there is some other semantics here you could investigate